src/Item.js

```diff
@@ -22,12 +24,29 @@ function Item(props) {
         'formData.purchaseDates': purchaseDates,
       });
     } else {
-      const date = new Date();
-      db.collection(userToken)
-        .doc(id)
-        .update({
-          'formData.purchaseDates': [...purchaseDates, date],
-        });
+      const newDate = new Date();
+      const newDates = [...purchaseDates, newDate];
+      const numberOfPurchases = newDates.length;
+      let latestInterval = null;
+
+      if (numberOfPurchases >= 2) {
```

Comment:
I think that this if check can be removed because the `estimates` function is already making this check and returning the correct value if there are less than two items, but it works either way 😃

Reply:
You can disregard this comment, I figured out why we need this if check now 👍🏼  Great job with all these calculations!